src/value-class.ts

```diff
@@ -3,8 +3,7 @@ import { NamedTypeDefinition } from './reflection-type';
 export type Branded<A, BrandTag> = A & Brand<BrandTag>;
 
 export class Brand<B> {
-  // @ts-ignore
-  private valueClassBrand: B; // branding, DO NOT ACCESS
+  public readonly valueClassBrand!: B; // branding, DO NOT ACCESS
```

Comment:
probably okay to make these public... 🤷‍♂️

Reply:
eh why? those are private for the express purpose of not allowing autocomplete

Reply:
TypeScript loses the type unless the property is public. Same goes for the `valueClassShape`. And if a generic type isn't used anywhere in a class, it gets lost and can't be inferred anywhere where that class is used.

*.d.ts
```ts
// when public
export declare class Brand<B> {
    readonly valueClassBrand: B;
}
// when private
export declare class Brand<B> {
    readonly valueClassBrand;
}
```

Reply:
Changed it to `protected` as it seems to keep the types in that case, but then it doesn't show up in autocomplete in most places 😬

Reply:
didn't we have this discussion a while ago?

Reply:
I remember talking about this before, yeah 😅